.circleci/config.yml

```diff
@@ -19,6 +19,7 @@ jobs:
             else
               python3 -m venv .venv
               source .venv/bin/activate
+              pip install --upgrade pip
```

Comment:
This needs a second and third pair of eyes.

Reply:
Why do we need this? I don't really know how to test this change, since it only runs in circleci

Reply:
Installing Celery will break with an error without it.

Reply:
Well, I guess it should work :shrug:

TBH, I don't even fully understand our config file. For example, line 8 says
```
- image: cimg/pytpip hon:3.11.7
```

But I cannot find information anywhere on what `cimg/pytpip` is. The circleci website only knows `cimg/python` and chatgpt thinks its a typo...